### PR TITLE
feat(clients): Implement Add method for V2 EventClient

### PR DESCRIPTION
**Pull request by FelixTing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the current behavior?

There is no V2 EventClient.

## What is the new behavior?
Added the V2 EventClient and implemented the Add method, which is used to issue an HTTP POST request to the core-data service to add new events. This PR also introduced the HTTP POST helper function for V2 clients.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Comment by FelixTing** [comment]
rebased

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/go-playground/validator/v10 v10.3.0
 	github.com/go-stack/stack v1.8.0 // indirect
 	github.com/google/uuid v1.1.2
-	github.com/stretchr/testify v1.5.1
+	github.com/stretchr/testify v1.6.1
```
Comment (cloudxxx8) [inline go.mod:10]: why do you need this upgrade?

File: v2/constants.go [hunk 2]
```diff
@@ -97,3 +97,9 @@ const (
 	DefaultLimit   = 20
 	CommaSeparator = ","
 )
+
+// Constants related to the possible content types supported by the APIs
+const (
```
Comment (cloudxxx8) [inline v2/constants.go:102]: use clients.ContentType, clients.ContentTypeJSON

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/clients/http/event.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+// NewEventClient creates an instance of EventClient
+func NewEventClient(baseUrl string) interfaces.EventClient {
+	return &eventClient{
+		baseUrl: baseUrl,
+	}
+}
+
+func (ec *eventClient) Add(ctx context.Context, events []models.Event) ([]common.BaseWithIdResponse, errors.EdgeX) {
```
Comment (cloudxxx8) [inline v2/clients/http/event.go:32]: the input signature should be AddEventRequest directly

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

File: v2/clients/http/utils/post.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+// Helper method to make the post JSON request and return the body
+func PostRequest(
+	ctx context.Context,
+	returnValuePointer interface{},
+	url string,
+	data interface{}) errors.EdgeX {
+
+	jasonEncodedData, err := json.Marshal(data)
```
Comment (lenny-intel) [inline v2/clients/http/utils/post.go:25]: Change `jasonEncodedData` => `jsonEncodedData`

File: v2/clients/http/utils/post.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	if err := json.Unmarshal(res, returnValuePointer); err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to parse the response body", err)
+	}
+	return nil
+
+}
+
+// PostRequestWithURL will make a POST request to the specified URL.
+// It returns the body as a byte array if successful and an error otherwise.
+func PostRequestWithURL(ctx context.Context, url string, data []byte) ([]byte, errors.EdgeX) {
```
Comment (lenny-intel) [inline v2/clients/http/utils/post.go:43]: Naming of these func needs tweaking. `PostRequestWithURL` and `PostRequest` both take urls. How about this one change to `PostRawDataRequest`

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/clients/http/utils/post.go (hunk 4 above)
Comment (FelixTing) [reply v2/clients/http/utils/post.go]: done

File: v2/clients/http/utils/post.go (hunk 5 above)
Comment (FelixTing) [reply v2/clients/http/utils/post.go]: Thanks @lenny-intel. `PostRawDataRequest` is really better than the old one.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by FelixTing** [comment]
rebased

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**